### PR TITLE
fix(pegasus): use dummy ICS20-1 transfer `sender`

### LINKED ISSUE #4498: Pegasus doesn't have a way to send the ICS20-1 required sender field
## Describe the bug
Sending an ICS20-1 packet the way Pegasus currently does results in an error.

## To Reproduce
1. Start an IBC relayer
2. Establish a connection and send a transfer packet in the REPL:
```
command[6] E(home.ibcport[0]).connect('/ibc-hop/connection-0/ibc-port/transfer/unordered/ics20-1')
history[6] [Object Alleged: Connection]{}
command[7] E(history[6]).send(JSON.stringify({ receiver: 'cosmos1dlua5s27mjvt0tedlfq03m3rc30asvt5wxt5s8', denom: 'hellocoin', amount: '100' }))
history[7] "{\"error\":\"sender address cannot be blank: invalid address\"}"
```
4. See the error above

## Expected behavior
Send of the transfer packet results in success message.

## Platform Environment
 - what OS are you using? what version of Node.js? irrelevant
 - is there anything special/unusual about your platform? no
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`) current master

## Additional context
I'm using the following branch of the Hermes relayer: https://github.com/agoric-labs/ibc-rs/tree/d082feae2c1a9925ec24fd4654d098eca91980e0

## Screenshots
image

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #4498

## Description

Implement a static string, `'pegasus'` for the transfer `sender` field, just to satisfy a needless assertion in `ibc-go`.

### Documentation Considerations

None.

### Testing Considerations

Better unit tests would be good, but the types guided this implementation.

### CODE REVIEW

File: packages/pegasus/src/courier.js [hunk 1]
```diff
@@ -54,13 +55,14 @@ export const makeCourierMaker =
     transferProtocol,
   }) => {
     /** @type {Sender} */
-    const send = async (zcfSeat, depositAddress) => {
+    const send = async (zcfSeat, depositAddress, refundAddress) => {
```
Comment (dckc) [inline packages/pegasus/src/courier.js]: Ah... `refundAddress` makes perfect sense.

I was going to ask about security considerations around "sender" and whether it's as forgeable as email sender. The result of [Joe-jobbing](https://en.wikipedia.org/wiki/Joe_job) here is that any refund goes to Joe, so there's no motivation for forgery.

Let's not call it `sender` anywhere (including the title of this PR) please.
_p.s. I see ICS20 calls it sender. Ugh._

File: packages/pegasus/src/ics20.js [hunk 2]
```diff
@@ -37,10 +37,11 @@ const safeJSONParseObject = s => {
  */
 export const parseICS20TransferPacket = async packet => {
   const ics20Packet = safeJSONParseObject(packet);
-  const { amount, denom, receiver } = ics20Packet;
+  const { amount, denom, receiver, sender } = ics20Packet;
```
Comment (dckc) [inline packages/pegasus/src/ics20.js]: The name here is not our choice, is it? Ugh. I see `sender` all over the ICS-20 spec. Sigh.

Comment (dckc) [reply packages/pegasus/src/ics20.js]: For the git headline, how about something like:

`feat(pegasus): allow refundAddress (aka sender) in transfer invitation`

Or is it `require ...`? I'm getting mixed signals. The `sender address cannot be blank` diagnostic in #4498 sure makes it look required.

File: packages/pegasus/src/ics20.js [hunk 3]
```diff
@@ -37,10 +37,11 @@ const safeJSONParseObject = s => {
...
 
   assert.typeof(denom, 'string', X`Denom ${denom} must be a string`);
   assert.typeof(receiver, 'string', X`Receiver ${receiver} must be a string`);
+  assert.typeof(sender, 'string', X`Sender ${sender} must be a string`);
```
Comment (dckc) [inline packages/pegasus/src/ics20.js]: `sender` is optional in the `ICS20TransferPacket` typedef. Why is it not optional here?

The ICS20 spec doesn't seem to help here. It uses golang types, where everything has a "zero" value, right?

**Review by dckc — COMMENTED** [review]
I need better clarity on whether `sender` is required.

And I prefer that we hoist `refundAddress` into the git headline.

**Comment by michaelfig** [comment]
Hi @dckc, can you PTA(nother)L?  The fix, as tested in the IBC meeting, is trivial.

File: packages/pegasus/src/ics20.js [hunk 4]
```diff
@@ -78,6 +83,7 @@ export const makeICS20TransferPacket = async ({
     amount: stringValue,
     denom: remoteDenom,
     receiver: depositAddress,
+    sender: OVERRIDE_SENDER_ADDRESS,
```
Comment (dckc) [inline packages/pegasus/src/ics20.js]:
```suggestion
    sender: DUMMY_SENDER_ADDRESS,
```

File: packages/pegasus/src/ics20.js [hunk 5]
```diff
@@ -8,10 +8,15 @@ import { assert, details as X } from '@agoric/assert';
...
  * @property {Denom} denom The denomination of the amount
- * @property {string} [sender] The sender address
+ * @property {string} sender The sender address
  * @property {DepositAddress} receiver The receiver deposit address
  */
 
+// ibc-go as late as v3 requires the `sender` to be nonempty, but doesn't
+// actually use it on the receiving side.  We don't need it on the sending side,
+// either, so we can just omit it.
+export const OVERRIDE_SENDER_ADDRESS = 'pegasus';
```
Comment (dckc) [inline packages/pegasus/src/ics20.js]: not sure it's worth another ci spin, but...

I don't know what we're overriding. "dummy" makes sense to me:

```suggestion
export const DUMMY_SENDER_ADDRESS = 'pegasus';
```

**Review by dckc — APPROVED** [review]
I have a bike-shedding suggestion.

But regardless, I approve.

**Pull request merged**